crates/rust-analyzer/src/main_loop.rs

```diff
@@ -625,6 +644,35 @@ impl GlobalState {
         }
     }
 
+    fn handle_queued_task(&mut self, task: QueuedTask) {
+        match task {
+            QueuedTask::CheckIfIndexed(uri) => {
+                let snap = self.snapshot();
+                let sender = self.sender.clone();
+
+                self.task_pool.handle.spawn_with_sender(ThreadIntent::Worker, move |_| {
+                    let _p = profile::span("GlobalState::check_if_indexed");
+                    tracing::debug!(?uri, "handling uri");
+                    let id = from_proto::file_id(&snap, &uri).expect("unable to get FileId");
+                    if let Ok(crates) = &snap.analysis.crates_for(id) {
+                        if crates.is_empty() {
+                            let not = lsp_server::Notification::new(
+                                ext::UnindexedProject::METHOD.to_string(),
+                                ext::UnindexedProjectParams {
+                                    text_documents: vec![lsp_types::TextDocumentIdentifier { uri }],
+                                },
+                            );
+                            sender.send(not.into()).expect("unable to send notification");
```

Comment:
Hm, instead of exposing the sender stuff here, I think I'd prefer sending back a `Task` (via the `_` parameter of the spawn closure, that's a task sender) and add a `Task::Notification` kind to be used for that. (then we can undo the `+ Send` changes and the `pub(crate)`)

Reply:
I wrote that initially (prior to publishing this PR), but opted for the current approach because I wasn’t sure if it’d be a premature architectural change. Will switch back!

Reply:
fixed!